### ISSUE TITLE
The "cal_entropy" function is missing in the "trainer.py"

### ISSUE THREAD

**Issue by AaronCheung524** [Davidcoach/RetouchFormer_AAAI_24#3]
In line **432** of [trainer.py](https://github.com/Davidcoach/RetouchFormer_AAAI_24/blob/main/core/trainer.py):

`entropy_loss = cal_entropy(head2_unpair_lq).mean()`

The function "cal_entropy" is not defined before. Is it implemented in this way?

```
def cal_entropy(p, eps=1e-12):
    p = torch.clamp(p, eps, 1.0)
    return -(p * torch.log(p))
```

**Comment by Davidcoach** [comment]
Here is our implementation:
def cal_entropy(probabilities, eps=1e-8):
    entropy = -probabilities * torch.log(probabilities + eps)
    return entropy.sum(dim=-1)

**Comment by AaronCheung524** [comment]
Thanks for your response. I will close this issue
